clash-cores/src/Clash/Cores/Xilinx/VIO/Internal/BlackBoxes.hs

```diff
@@ -168,17 +187,17 @@ vioProbeBBTF bbCtx
                 _         -> pure <$> DSL.assign (head inNames) singleInput
             _ -> zipWithM DSL.assign inNames inPs
 
-        outProbes <-
-          forM (zip outNames outTys) $ uncurry DSL.declare
-        outProbesBV <-
-          forM (zip outputNames outProbes) $ uncurry (DSL.toBvWithAttrs attrs)
+        outProbes <- zipWithM DSL.declare outNames outTys
+        outProbesBV0 <- zipWithM (DSL.toBvWithAttrs attrs) userOutputNames outProbes
+        inProbesBV0 <- zipWithM (DSL.toBvWithAttrs attrs) userInputNames inProbes
```

Comment:
This introduces a lot of uncessary names. Why not
```haskell
  ...
  inProbesBV <- zipWithM toNameCheckedBv userInputNames inProbes
  ...
where
  toNameCheckedBv x =
    -- The HDL attribute 'KEEP' is added to the signals connected to the
    -- probe ports so they are not optimized away by the synthesis tool.    
    fmap (checkNameCollision x) . DSL.toBvWithAttrs [StringAttr' "KEEP" "true"] x
```
which also avoids the introduction of `attrs`.

Reply:
Good idea!